zcash_client_backend/proto/compact_formats.proto

```diff
@@ -33,16 +41,30 @@ message CompactTx {
     //    valueBalance + (sum(vPubNew) - sum(vPubOld) - sum(tOut))
     uint32 fee = 3;
 
-    repeated CompactSpend spends = 4;
-    repeated CompactOutput outputs = 5;
+    repeated CompactSaplingSpend spends = 4;
+    repeated CompactSaplingOutput outputs = 5;
+    repeated CompactOrchardAction actions = 6;
+}
+
+// CompactSaplingSpend is a Sapling Spend Description as described in 7.3 of the Zcash
+// protocol specification.
+message CompactSaplingSpend {
+    bytes nf = 1;   // nullifier (see the Zcash protocol specification)
 }
 
-message CompactSpend {
-    bytes nf = 1;
+// output is a Sapling Output Description as described in section 7.4 of the
+// Zcash protocol spec. Total size is 948.
+message CompactSaplingOutput {
+    bytes cmu = 1;          // note commitment u-coordinate
+    bytes epk = 2;          // ephemeral public key
+    bytes ciphertext = 3;   // first 52 bytes of ciphertext
 }
 
-message CompactOutput {
-    bytes cmu = 1;
-    bytes epk = 2;
-    bytes ciphertext = 3;
+// https://github.com/zcash/zips/blob/main/zip-0225.rst#orchard-action-description-orchardaction
+// (but not all fields are needed)
+message CompactOrchardAction {
+    bytes nullifier = 1;        // [32] The nullifier of the input note
+    bytes cmx = 2;              // [32] The x-coordinate of the note commitment for the output note
+    bytes ephemeralKey = 3;     // [32] An encoding of an ephemeral Pallas public key
+    bytes encCiphertext = 4;    // [580] The encrypted contents of the note plaintext
```

Comment:
@LarryRuane why was this field documented in https://github.com/zcash/lightwalletd/pull/371 as being the full 580 bytes? We only plan to send the first 52 bytes of it, i.e. as documented in https://zips.z.cash/zip-0244#t-4a-orchard-actions-compact-digest

```suggestion
    bytes ciphertext = 4;       // [52] The note plaintext component of the encCiphertext field
```

Reply:
Ah yes, I vaguely remember that at one time the lightwalletd code thought that this field of the compact format comprised all 580 bytes, then I discovered that's incorrect, but I forgot to update this comment. I just went ahead and merged this PR to fix the comment: https://github.com/zcash/lightwalletd/pull/377 so please pull a copy of `compact_formats.proto` from master again.

Reply:
The suggested description is not quite accurate, but "// first 52 bytes of ciphertext" as used in https://github.com/zcash/lightwalletd/pull/377 is.

("Note plaintext" includes the memo; also, it's not accurate to refer to components of `encCiphertext`.)